### PR TITLE
 first try to kill all mutant

### PR DESCRIPTION
**Pull request by fezfez**
as proposed before, this is a solution to kill all mutant and have 100% coverage.

still WIP

### CODE REVIEW

**Comment by guillaume-sainthillier** [comment]
I must admit I don't really see the benefits of this pattern. IMHO it's lowering the general readability.
But as you add more tests to this library I could not deny the merge :white_check_mark:

File: src/Parser/ParseAmount.php [hunk 1]
```diff
@@ -0,0 +1,52 @@
+<?php
+
+
+namespace Silarhi\Cfonb\Parser;
+
+
+use Silarhi\Cfonb\Exceptions\ParseException;
+
+class ParseAmount
```
Comment (guillaume-sainthillier) [inline src/Parser/ParseAmount.php:9]: Could you please rename `ParseAmount` class to `AmountParser`?

File: src/Parser/ParseDate.php [hunk 2]
```diff
@@ -0,0 +1,20 @@
+<?php
+
+
+namespace Silarhi\Cfonb\Parser;
+
+
+use Silarhi\Cfonb\Exceptions\ParseException;
+
+class ParseDate
```
Comment (guillaume-sainthillier) [inline src/Parser/ParseDate.php:9]: Could you please rename `ParseDate` class to `DateParser`?

**Review by guillaume-sainthillier — COMMENTED** [review]
I an general way, could you rename `__invoke` magic method to a `parse` method?
I'm sorry but I don't really see the benefit of using this magic method over a custom one, especially i you explicitly call the `$object->__invoke(...)` .

**Comment by guillaume-sainthillier** [comment]
Anyway thank you for adding test to this lib

**Comment by fezfez** [comment]
> I an general way, could you rename `__invoke` magic method to a `parse` method?
> I'm sorry but I don't really see the benefit of using this magic method over a custom one, especially i you explicitly call the `$object->__invoke(...)` .

mainly inspired by @ocramius, i generally use __invoke instead of "**insert method name**" because

- invokable class will never have another public method, they contains only one method
- this force you to have a clear class name
- there are like function but with DI
- they have no side effect

see discussion : 

- https://twitter.com/ocramius/status/621649385345150976
- https://gist.github.com/Ocramius/b3a5c4b5610efc132411

I can change `(new AmountParser())->__invoke()` to `(new AmountParser())->parse()` if you always want it  :rainbow: :nerd_face:

**Comment by guillaume-sainthillier** [comment]
Thank you for explanation :-)
I still prefer to use "explicit" method name, but it will keep some of "invokable" pattern benefits:

1. This force you to have a clear class name
2. There are like function but with DI
3. Unit testable
4. Decoupled from business logic

File: src/Parser/ParseAmount.php (hunk 1 above)
Comment (fezfez) [reply src/Parser/ParseAmount.php]: done

File: src/Parser/ParseDate.php (hunk 2 above)
Comment (fezfez) [reply src/Parser/ParseDate.php]: done

**Comment by fezfez** [comment]
@guillaume-sainthillier  all mutant are killed ;)

**Comment by guillaume-sainthillier** [comment]
Thank you Stéphane.

**Pull request merged**